src/server/ocpp/utils/OCPPUtils.ts

```diff
@@ -331,14 +331,21 @@ export default class OCPPUtils {
               lastUpdate: new Date()
             };
           } catch (error) {
+            const message = `Billing - startTransaction failed - transaction ID '${transaction.id}'`;
             await Logging.logError({
               tenantID: tenantID,
               user: transaction.userID,
               source: Constants.CENTRAL_SERVER,
               action: ServerAction.BILLING_TRANSACTION,
               module: MODULE_NAME, method: 'billTransaction',
-              message: `Failed to bill the Transaction ID '${transaction.id}'`,
-              detailedMessages: { error: error.message, stack: error.stack }
+              message, detailedMessages: { error: error.message, stack: error.stack }
+            });
+            // Prevent from starting a transaction when Billing prerequisites are not met
+            throw new BackendError({
+              user: transaction.user,
+              action: ServerAction.BILLING_TRANSACTION,
+              module: MODULE_NAME, method: 'billTransaction',
+              message, detailedMessages: { error: error.message, stack: error.stack }
```

Comment:
Could you please explain us the conditions that trigger the error throwing (to ease the PR review without proper `git checkout` on the branch)?

I think error throwing is fine to stop the start transaction, maybe a bit brutal. 
Can you return a boolean in that static method that will tell if the transaction is allowed or not?
The method will keep logging the details and the caller will then be able report a short message to the end users. Maybe you can also return an object like `{ cardSubcriptionError: true, updateTransactionError: false, etc. }` to permit to display more accurate error messages to end users

Reply:
No 😔

Reply:
It's rejected and we check the logs.

Reply:
I think so, the error message reporting refinement can be done later one.